### PR TITLE
[USER32_APITEST] Add test for bad 'Icon Groups' but good 'Icons' such as F1'97

### PR DESCRIPTION
**Pull request by Doug-Lyons**
## Purpose

_Add regression test for these special cases such as those for F1'97_
_These cases can be created by old versions of Watcom C/C++_
_They show their icons in Windows Explorer WinXP/Win2K3, but not Win7_

JIRA issue: [CORE-10726](https://jira.reactos.org/browse/CORE-10726)

## Proposed changes

_Add code to find first 'Icon' if finding 'Icon Groups' fails to work correctly._

### CODE REVIEW

**Review by JoachimHenze — APPROVED** [review]

File: modules/rostests/apitests/user32/PrivateExtractIcons.c [hunk 1]
```diff
@@ -90,6 +91,12 @@ static struct
 
     /* Icon group file containing one PNG icon and one normal icon */
     {L"%temp%\\ROS.ico", 1, 1, TRUE},
+
+    /* Executable file with bad 'Icon Group' but good 'Icons'.
+     * Windows explorer shows the program's icon correctly in WinXP/Win2K3
+     * but Windows 7 shows only a default icon. This is analogous
```
Comment (SergeGautherie) [inline modules/rostests/apitests/user32/PrivateExtractIcons.c:97]: Would someone know what Vista shows?

Comment (JoachimHenze) [reply modules/rostests/apitests/user32/PrivateExtractIcons.c:97]: I am not aware of any active ros developer operating (or owning) that OS. I think it is acceptable to add that information whenever the need would evolve. We already do know that different versions of Windows do behave differently here, that is what I would consider to be most important for now. After all it is most likely that if we ever aimed to retarget ros in the future, that we will entirely skip the Vista-intermediate-step and jump directly to 7 or even higher.
Unless you could provide the answer yourself, I don't think it makes sense to postpone merging the test just due to that unresolved question.

Comment (binarymaster) [reply modules/rostests/apitests/user32/PrivateExtractIcons.c:97]: We have testers with Vista VMs, just ask if needed.

File: modules/rostests/apitests/user32/PrivateExtractIcons.c [hunk 2]
```diff
@@ -27,8 +27,9 @@ BOOL ResourceToFile(INT i, PCWSTR FileName)
 
     if (FileExists(FileName))
     {
-        skip("'%S' already exists. Exiting now\n", FileName);
-        return FALSE;
+        /* We should only be using %temp% paths, so deleting here should be OK */
+        DeleteFileW(FileName);
+        printf("'%S' already exists. Deleting now\n", FileName);
```
Comment (binarymaster) [inline modules/rostests/apitests/user32/PrivateExtractIcons.c:32]:
```suggestion
        printf("Deleting '%S' that already exists.\n", FileName);
        DeleteFileW(FileName);
```

File: modules/rostests/apitests/user32/PrivateExtractIcons.c (hunk 1 above)
Comment (Doug-Lyons) [reply modules/rostests/apitests/user32/PrivateExtractIcons.c:97]: I just installed Windows Vista and it only shows the default icon.

File: modules/rostests/apitests/user32/PrivateExtractIcons.c (hunk 2 above)
Comment (Doug-Lyons) [reply modules/rostests/apitests/user32/PrivateExtractIcons.c]: Changed. Thanks.

File: modules/rostests/apitests/user32/PrivateExtractIcons.c (hunk 1 above)
Comment (SergeGautherie) [inline modules/rostests/apitests/user32/PrivateExtractIcons.c:97]:
```suggestion
     * but WinVista/Win7 show only a default icon. This is analogous
```

File: modules/rostests/apitests/user32/PrivateExtractIcons.c [hunk 3]
```diff
@@ -90,6 +91,12 @@ static struct
 
     /* Icon group file containing one PNG icon and one normal icon */
     {L"%temp%\\ROS.ico", 1, 1, TRUE},
+
+    /* Executable file with bad 'Icon Group' but good 'Icons'.
+     * Windows explorer shows the program's icon correctly in WinXP/Win2K3
+     * but Windows 7 shows only a default icon. This is analogous
+     * to EXE's generated by older Watcom C/C++ versions. */
```
Comment (SergeGautherie) [inline modules/rostests/apitests/user32/PrivateExtractIcons.c:98]:
```suggestion
     * to EXE's generated by older Watcom C/C++ versions */
```

File: modules/rostests/apitests/user32/PrivateExtractIcons.c [hunk 4]
```diff
@@ -27,8 +27,9 @@ BOOL ResourceToFile(INT i, PCWSTR FileName)
 
     if (FileExists(FileName))
     {
-        skip("'%S' already exists. Exiting now\n", FileName);
-        return FALSE;
+        /* We should only be using %temp% paths, so deleting here should be OK */
+        printf("Deleting '%S' that already exists.\n", FileName);
```
[the remark this answers is not in the document]
Comment (Doug-Lyons) [reply modules/rostests/apitests/user32/PrivateExtractIcons.c:31]: I changed this just a bit ago to match a request from @binarymaster. Let's leave it like this for now. Thanks.

File: modules/rostests/apitests/user32/PrivateExtractIcons.c (hunk 3 above)
Comment (Doug-Lyons) [reply modules/rostests/apitests/user32/PrivateExtractIcons.c:98]: The period here is not something that will add to a 'debug.log', so it should be OK. Thanks.

File: modules/rostests/apitests/user32/PrivateExtractIcons.c (hunk 1 above)
Comment (Doug-Lyons) [reply modules/rostests/apitests/user32/PrivateExtractIcons.c:97]: I want to leave this and only make this change is other changes are needed. Thanks.

**Comment by HBelusca** [comment]
@Doug-Lyons Do you have screenshots that show the tests passing on Windows?

**Comment by Doug-Lyons** [comment]
> @Doug-Lyons Do you have screenshots that show the tests passing on Windows?

See Below:

User32_apitest_PrivateExtratIcons_W2K3SP2a

**Comment by Doug-Lyons** [comment]
Thanks, @HBelusca and @JoachimHenze for your help with getting this approved and committed.

**Pull request merged**